11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py

```diff
@@ -0,0 +1,232 @@
+    random_state=0,
+)
+
+# verify that the correct splits have been made
+# 100% test set
+print(f"Shape for the 100% test set: {test_set_all.shape}\n")
+
+# 75% test set and 25% train set
+print(
+    f"Shape for the 75% test set: {training_data_set_75.shape};\nShape for the 75% train set: {testing_data_set_75.shape}\n"
+)
+
+# 50% test set and 50% train set
+print(
+    f"Shape for the 50% test set: {training_data_set_50.shape};\nShape for the 50% train set: {testing_data_set_50.shape}"
+)
+
+
+# In[10]:
+
+
+# combine all testing sets together while preserving the index
+testing_data_set = pd.concat(
+    [test_set_all, testing_data_set_75, testing_data_set_50], axis=0
+)
+testing_data_set = testing_data_set.sort_index()
+testing_data_set
+
+# combine all training sets together while preserving the index
+training_data_set = pd.concat([training_data_set_75, training_data_set_50], axis=0)
+training_data_set = training_data_set.sort_index()
+training_data_set
+
+print(
+    f"Testing set length: {len(testing_data_set)}\nTraining set length: {len(training_data_set)}"
+)
+
+# get the indexes for the training and testing sets
+testing_data_set_index = testing_data_set.index
+training_data_set_index = training_data_set.index
+
+
+# In[11]:
+
+
+# create pandas dataframe with all indexes and their respective labels, stratified by phenotypic class
+index_data = []
+for index in training_data_set_index:
+    index_data.append({"labeled_data_index": index, "label": "train"})
+for index in testing_data_set_index:
+    index_data.append({"labeled_data_index": index, "label": "test"})
+
+# make index data a dataframe and sort it by labeled data index
+index_data = pd.DataFrame(index_data).sort_values(["labeled_data_index"])
+
+
+# In[12]:
+
+
+# set save path
+if aggregation == True:
+    if nomic == True:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+    elif nomic == False:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+elif aggregation == False:
+    if nomic == True:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+    elif nomic == False:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+else:
+    print("Error")
+
+print(save_path)
+# create save path if it doesn't exist
+save_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[13]:
+
+
+# save indexes as tsv file
+if aggregation == True:
+    if nomic == True:
+        index_data.to_csv(
+            f"{save_path}/aggregated_sc_and_nomic_data_split_indexes.tsv", sep="\t"
+        )
+    elif nomic == False:
+        index_data.to_csv(f"{save_path}/aggregated_sc_data_split_indexes.tsv", sep="\t")
+elif aggregation == False:
+    if nomic == True:
+        index_data.to_csv(f"{save_path}/sc_and_nomic_data_split_indexes.tsv", sep="\t")
+    elif nomic == False:
+        index_data.to_csv(f"{save_path}/sc_split_indexes.tsv", sep="\t")
+else:
+    print("Error")
```

Comment:
this is a non-specific error message that is not informative if anyone gets to this

Reply:
Made this error message more specific